src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1630,6 +1633,112 @@ export class ResultsViewPageStore {
         }
     });
 
+    /**
+     * Fetch fusion data from the backend API with following request parameters:
+     * - entrez gene ids
+     * - sample molecular identifiers
+     * - molecular profile ids (only when sample molecular identifiers are not provided)
+     * @type {MobxPromiseUnionType<StructuralVariant[]>}
+     */
+    readonly fusions = remoteData<StructuralVariant[]>({
+        await: () => [
+            this.genes,
+            this.selectedMolecularProfiles,
+            this.samples,
+            this.studyIdToStudy
+        ],
+        invoke: async () => {
+            let molecularProfiles: Array<string> = [];
+            let samples: SampleMolecularIdentifier[] = [];
+            let params: StructuralVariantFilterExt = {entrezGeneIds:[]};
+            if (this.genes.result) {
+                params.entrezGeneIds = _.map(this.genes.result, (gene: Gene) => gene.entrezGeneId);
+            }
+            // compose request parameters
+            if (this.samples.result) {
+                // get sample molecular identifiers
+                samples = this.samples.result.map(sample => {
+                    return {
+                        molecularProfileId: sample.studyId + '_structural_variants',
+                        sampleId: sample.sampleId
+                    };
+                });
+                params.sampleMolecularIdentifiers = samples;
+            } else {
+                // get molecular profile ids
+                if (this.selectedMolecularProfiles.result) {
+                    molecularProfiles = this.selectedMolecularProfiles.result.filter(profile => {
+                        return profile.molecularAlterationType === AlterationTypeConstants.STRUCTURAL_VARIANT;
+                    }).map(profile => {
+                        return profile.molecularProfileId;
+                    });
+                    params.molecularProfileIds = molecularProfiles;
+                }
+            }
+            return await client.fetchStructuralVariantsUsingPOST({'structuralVariantFilter':<StructuralVariantFilter> params});
```

Comment:
await is only necessary when you want to assign the asyncronous result to a local variable and "pretend" to program sychronously.   here, you are just returning a promise, so you can take the async tag off the invoke function and just return client.fetch___